GitHub.Collectors.Functions/GitHubFunctions.cs

```diff
@@ -706,6 +710,116 @@ public async Task ExecutePointCollectorAsync(string queueItem, ExecutionContext
+                foreach (JObject installation in installations)
+                {
+                    JObject formattedInstallation = new JObject();
+                    formattedInstallation.Add("OrganizationLogin", installation.SelectToken("account.login"));
+                    formattedInstallation.Add("OrganizationId", installation.SelectToken("account.id"));
+                    formattedInstallations.Add(formattedInstallation);
+
+                    int id = installation.SelectToken("account.id").Value<int>();
+                    string login = installation.SelectToken("account.login").Value<string>();
+
+                    discoveredOrganizationMap[id] = login;
+                }
+
+                formattedInstallations = new JArray(formattedInstallations.OrderBy(obj => (string)obj["OrganizationLogin"]));
+                string configurationString = formattedInstallations.ToString();
+
+                // output to blob.
+                CloudBlockBlob outputBlob = AzureHelpers.GetBlob("github-settings", "generated-organizations.json");
+                CloudBlobStream cloudBlobStream = await outputBlob.OpenWriteAsync().ConfigureAwait(false);
+                StreamWriter writer = new StreamWriter(cloudBlobStream, Encoding.UTF8);
+                await writer.WriteLineAsync(configurationString).ConfigureAwait(false);
+                writer.Dispose();
+
+                // find organization diff.
+                string organizations = await AzureHelpers.GetBlobContentAsync("github-settings", "organizations.json").ConfigureAwait(false);
+                JArray organizationsArray = JArray.Parse(organizations);
+
+                foreach (JObject organization in organizationsArray)
+                {
+                    int id = organization.SelectToken("OrganizationId").Value<int>();
+                    if( discoveredOrganizationMap.ContainsKey(id) )
+                    {
+                        discoveredOrganizationMap.Remove(id);
+                    }
+                }
+
+                // add messages to auto-onbaording queue (moved manually to onboarding after PR is merged)
+                foreach (int id in discoveredOrganizationMap.Keys )
+                {
+                    OnboardingInput onboardingInput = new OnboardingInput()
+                    {
+                        OnboardingType = OnboardingType.Repository,
+                        OrganizationId = id,
+                        OrganizationLogin = discoveredOrganizationMap[id]
+                    };
+
+                    CloudQueue onboardingCloudQueue = await AzureHelpers.GetStorageQueueAsync("onboarding-auto").ConfigureAwait(false);
+                    IQueue onboardingQueue = new CloudQueueWrapper(onboardingCloudQueue);
+                    await onboardingQueue.PutObjectAsJsonStringAsync(onboardingInput, TimeSpan.MaxValue).ConfigureAwait(false);
+                }
```

Comment:
What happens if we don't auto-onbaord for, let's say a week? Will it keep adding the same message (in the diff) every single day?

Reply:
If that is the case, is there a way to ensure that a message for a given (new) organization is only added once? Alternatively, I guess we could set message expiry to 1-day so that by the next run, the previous one expires? Not sure, just some suggestions.

Reply:
I have addressed this by reading from generated-organizations.json if it exists to calculate the diff. Then repeat organizations will not be added to the onboarding-auto queue.